iothub/device/src/Common/Exceptions/IotHubException.cs

```diff
@@ -139,11 +129,26 @@ protected IotHubException(SerializationInfo info, StreamingContext context)
             }
         }
 
+        internal IotHubException(bool isTransient) : base()
+        {
+            IsTransient = isTransient;
+        }
+
+        /// <summary>
+        /// Indicates if the error is transient and should be retried.
+        /// </summary>
+        public bool IsTransient { get; private set; }
+
+        /// <summary>
+        /// The service returned tracking Id associated with this particular error.
+        /// </summary>
+        public string TrackingId { get; set; }
+
         /// <summary>
-        /// Sets the SerializationInfo object.
+        /// Sets the <see cref="IsTransient"/> and <see cref="TrackingId"/> information to the <see cref="SerializationInfo"/>.
```

Comment:
I don't quite get this sentence.

Reply:
What I meant here is, using this method you can set the tracking Id and transient flag into the serialization info for this exception. 
I can see how it can be confusing, esp since the method is called "Getxxx" and we perform a set operation within it: https://docs.microsoft.com/en-us/dotnet/api/system.exception.getobjectdata?view=net-5.0#System_Exception_GetObjectData_System_Runtime_Serialization_SerializationInfo_System_Runtime_Serialization_StreamingContext_

How about:
`Sets the <see cref="IsTransient"/> and <see cref="TrackingId"/> information to the <see cref="SerializationInfo"/> and performs serialization.`

Reply:
Wow the method really is very poorly named, but it is specified by `System.Exception`.

Why not just copy the doc comment from the base method, but indicate that is how one would include the additional properties defined in this exception: `IsTransient` and `TrackingId`?

Reply:
sure